### DISCUSSION TITLE
Multilingual support

### DISCUSSION THREAD

**Discussion by niral098** [GetDKAN/dkan#4167] [General]
Hi! Thank you for your hard work, guys! It's good to see the ongoing progress of Dkan. 
We are working on an opendata Dkan app with React frontend(D10) and have a question regarding its multilingual support(Spanish language). We want translations of /home, /datasets, /publishers, /about. Is it like we need to define custom translation pages within React? And is it possible to translate individual datasets(content data) with the help of Drupal's own content translation? Thank you!

**Comment by dafeder** [comment]
Hi @niral098, just want to be clear, are you talking here about simply having a Spanish-language site or having a _multi_ lingual site in Spanish and English/some other language?

I will say that we have unfortunately not done much testing yet with datasets and content translation. There are DKAN sites out there in other languages but none I know of yet in _multiple_ languages. The DKAN API for instance has no provisions for multiple translations of the same content; it assumes everything is in the site's default language. Certainly it is a gap that needs to be addressed.

**Comment by dgading** [comment]
I have not tried translating individual datasets, since that data would be coming from the metastore API so I don't have solution for that. 

For static pages, we did start a PR a couple of years ago, but it never was fully implemented. (https://github.com/GetDKAN/data-catalog-app/pull/100/files) using https://react.i18next.com/. With this library, you can create use their language detector and its hook will pull the correct text from the translation in the i18n.js file.

**Comment by niral098** [comment]
Hi @dafeder @dgading  I was talking about the multi-lingual feature. I understand that it is not supported at this point. But, I will look into the i18next thing for sure. Thank you for your time and support.

**Comment by dafeder** [comment]
No problem! Would be great to hear more about your project and if there's other ways we can support.

**Reply by niral098** [reply]
@dafeder Thank you! Your support means a lot to us.

**Comment by dafeder** [comment]
Also, there is now a multilingual DKAN 2 site live that I was previously unaware of! https://datos.cadiz.es/. 

See PRs against our FE repos to bring in multilingual functionality:

https://github.com/GetDKAN/data-catalog-components/pull/164
https://github.com/GetDKAN/data-catalog-app/pull/100

We are overdue to give these a close review but trying to prioritize better i18n support this year.

**Reply by niral098** [reply]
Hi @dafeder Thank you very much. I will check the PRs.
